src/Geoscape/DogfightState.cpp

```diff
@@ -1520,13 +1521,35 @@ void DogfightState::update()
 						}
 					}
 				}
-				if (!_state->getGlobe()->insideLand(_ufo->getLongitude(), _ufo->getLatitude()))
+				bool surviveSplashLand = false;
+				bool fakeUnderwaterTexture = _state->getGlobe()->insideFakeUnderwater(_ufo->getLongitude(), _ufo->getLatitude());
+				if (fakeUnderwaterTexture)
+				{
+					surviveSplashLand = getSurviveSplash(_ufo);
+				}
+				if (!_state->getGlobe()->insideLand(_ufo->getLongitude(), _ufo->getLatitude()) || !surviveSplashLand)
 				{
 					_ufo->setStatus(Ufo::DESTROYED);
 					_destroyUfo = true;
+					if (!fakeUnderwaterTexture) //special care for vanilla behavior in crash landing above water
+					{
+						setStatus("STR_UFO_CRASH_LANDS");
+					}
+					else
+					{
+						setStatus("STR_UFO_DESTROED_BY_SPLASHDOWN"); //separate case to make more it clear to the player
+					}
 				}
 				else
 				{
+					if (surviveSplashLand)
```

Comment:
fixed in commit 446bf6f. was changed to 
"if (fakeUnderwaterTexture && surviveSplashLand)"